junicode: 1.003 -> 2.200

ivan-timokhin opened the pull request:
## Description of changes

Release announcement:
https://github.com/psb1558/Junicode-font/releases/tag/v2.001

This is a breaking change, at least in font file naming (Junicode.ttf is now Junicode-Regular.ttf). In general, 2.0 adds a lot more font variants and opentype and web font versions of the font.

Seeing as backward compatibility is broken anyway, I opted to break it a bit more and change custom install path (`junicode-ttf`) to seemingly more conventional `truetype`; new .otf and .woff2 variants are then naturally placed in corresponding directories. This does *not* affect the `fonts.packages` NixOS option, which rearranges font files anyway, but brings a degree of consistency with other fonts.

Both the file renaming and the directory structure change break satysfi, however, so I adjusted its builder accordingly, copying over only those font variants that were also present in 1.0 series.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

ivan-timokhin committed:
junicode: 1.003 -> 2.001

Release announcement:
https://github.com/psb1558/Junicode-font/releases/tag/v2.001

This is a breaking change, at least in font file naming (Junicode.ttf
is now Junicode-Regular.ttf). In general, 2.0 adds a lot more font
variants and opentype and web font versions of the font.

Seeing as backward compatibility is broken anyway, I opted to break it
a bit more and change custom install path (`junicode-ttf`) to
seemingly more conventional `truetype`; new .otf and .woff2 variants
are then naturally placed in corresponding directories. This
does *not* affect the `fonts.packages` NixOS option, which rearranges
font files anyway, but brings a degree of consistency with other
fonts.

Both the file renaming and the directory structure change break
satysfi, however, so I adjusted its builder accordingly, copying over
only those font variants that were also present in 1.0 series.

ivan-timokhin commented on pkgs/tools/typesetting/satysfi/default.nix:
```diff
@@ -65,10 +65,13 @@ in
+      cp ${junicode}/share/fonts/truetype/Junicode-{Bold,BoldItalic,Italic}.ttf \
+        $out/share/satysfi/dist/fonts/
+      cp ${junicode}/share/fonts/truetype/Junicode-Regular.ttf \
+        $out/share/satysfi/dist/fonts/Junicode.ttf
```
@mt-caret @marsam
Would appreciate your thoughts on SATySFi changes.

h7x4 commented on pkgs/data/fonts/junicode/default.nix:
```diff
@@ -1,24 +1,19 @@
-    cp legacy/*.ttf $out_ttf
-
-    runHook postInstall
+    install -Dt $out/share/fonts/truetype TTF/*.ttf VAR/*.ttf
+    install -Dt $out/share/fonts/opentype OTF/*.otf
+    install -Dt $out/share/fonts/woff2 WOFF2/*.woff2
+    install -Dt $out/share/doc/${pname}-${version} docs/*.pdf
```
```suggestion
    runHook preInstall

    install -Dt $out/share/fonts/truetype TTF/*.ttf VAR/*.ttf
    install -Dt $out/share/fonts/opentype OTF/*.otf
    install -Dt $out/share/fonts/woff2 WOFF2/*.woff2
    install -Dt $out/share/doc/${pname}-${version} docs/*.pdf
    
    runHook postInstall
```

ivan-timokhin replied:
Fixed

h7x4 commented on pkgs/tools/typesetting/satysfi/default.nix:
```diff
@@ -65,10 +65,13 @@ in
         ${lmodern}/share/fonts/opentype/public/lm/* \
         ${lmmath}/share/fonts/opentype/latinmodern-math.otf \
         $out/share/satysfi/dist/fonts
+      cp ${junicode}/share/fonts/truetype/Junicode-{Bold,BoldItalic,Italic}.ttf \
```
This line could probably stay part of the previous `cp`

h7x4 commented on pkgs/data/fonts/junicode/default.nix:
```diff
@@ -1,24 +1,19 @@
+    install -Dt $out/share/fonts/truetype TTF/*.ttf VAR/*.ttf
+    install -Dt $out/share/fonts/opentype OTF/*.otf
+    install -Dt $out/share/fonts/woff2 WOFF2/*.woff2
+    install -Dt $out/share/doc/${pname}-${version} docs/*.pdf
```
Maybe the docs could be part of `$doc` instead?

ivan-timokhin replied:
That does seem better, yes. I blindly copied this part from other font derivations, most of which seem to put documentation in `$out`, but probably they are the ones in the need of refactoring. I’m not even sure if this documentation actually ends up anywhere in a NixOS install, as `fonts.fonts` seems to ignore it.

ivan-timokhin committed:
junicode: 2.001 -> 2.002

Release announcement:
https://github.com/psb1558/Junicode-font/releases/tag/v2.002

From it:

Version 2.002 adds U+0243 B with stroke (with small cap and petite cap
variants) and variants of U+014A Eng (with small cap and petite cap
variants). It also adds many anchors and refines and corrects many
outlines (especially in roman).

ivan-timokhin committed:
junicode: 2.002 -> 2.003

Release announcement:
https://github.com/psb1558/Junicode-font/releases/tag/v2.003

From it:

This is mostly a bug-fix version. It (1) fixes several non-functioning
tag sequences; (2) fixes an incompatibility with InDesign that caused
OpenType features to misbehave in that program; (3) improved several
outlines and adjusted badly positioned anchors.

ivan-timokhin committed:
junicode: 2.003 -> 2.004

Release announcement:
https://github.com/psb1558/Junicode-font/releases/tag/v2.004

From it:

This release includes (mainly in the italic face) revisions of metrics
and kerning of numbers and of kerning around parentheses and
brackets. It also fixes some spacing problems (in the roman face) with
capitals in parentheses (U+1F110-U+1F129) and a few problems with
disordered or wrong components.

h7x4 commented:
@ivan-timokhin Would you mind clearing up the merge conflict? I think this is ready to merge elsewise

ivan-timokhin committed:
junicode: 2.004 -> 2.100

Release announcement:
https://github.com/psb1558/Junicode-font/releases/tag/v2.100

From it:

Version 2.100 features a thorough reworking of the font's number
system. The default numbers are now oldstyle (or "lowercase")
proportional, and the metrics, kerning, and often the outlines of the
number glyphs have been refined, along with those of glyphs typically
associated with numbers (e.g. currency symbols). An alternate, more
modern style of some oldstyle numbers has been supplied and made
available via Stylistic Set 9 (ss09).

ivan-timokhin commented:
@h7x4 thank you for the reminder.

ivan-timokhin committed:
junicode: 2.100 -> 2.200

Release announcement:
https://github.com/psb1558/Junicode-font/releases/tag/v2.200

From it:

Junicode 2.200 features miscellaneous fixes and additions, and it also
inaugurates a long-term program of adding glyphs for the benefit of
the Ansund HTR (Handwritten Text Recognition) project, which is
developing tools for the automated reading of medieval manuscripts. Of
the additions made so far, the ones likely to be of greatest interest
to users are two series of medieval capitals, available via features
ss11[1] and ss11[2]. The first series consists of rustic capitals,
often used for text in late ancient and early medieval times and for
rubrics (headings) in the central Middle Ages. The second consists of
Lombardic capitals, used in the central and later Middle Ages for what
are now called drop caps. These capitals are designed to harmonize
with Junicode to the greatest degree possible while remaining faithful
to the medieval sources.
